amgut/lib/data_access/survey.py

```diff
@@ -282,52 +294,46 @@ def fetch_survey(self, survey_id):
-                    on sa.survey_question_id = sq.survey_question_id
-            where sa.survey_id = %s and sq.retired = FALSE""".format(
-            self._survey_answers_table,
-            self._survey_question_response_table,
-            self._survey_question_response_type_table,
-            self._questions_table),
-            [survey_id])
-
-        answers_other = db_conn.execute_fetchall("""
-            select survey_question_id, response
-            from {0}
-            left join {1} using (survey_question_id)
-            where survey_id = %s and retired = FALSE""".format(
-            self._survey_answers_other_table, self._questions_table),
-            [survey_id])
-
-        survey = defaultdict(list)
-        for qid, idx, qtype in answers:
-            eid = self.questions[qid].interface_element_ids[0]
-            if type(idx) in {str, unicode}:
-                idx = idx.encode('utf-8')
-            if qtype == 'SINGLE':
-                survey[eid] = idx
-            else:
-                survey[eid].append(idx)
-
-        for qid, data in answers_other:
-            eid = self.questions[qid].interface_element_ids[0]
-            data = str(data[2:-2])  # drop [""]
-            survey[eid] = data
-
-        if len(survey) == 0:
-            raise ValueError("Survey answers do not exist in DB: %s" %
-                             survey_id)
-        return survey
+        with TRN:
+            sql = """SELECT survey_question_id, display_index,
+                            survey_response_type
+                     FROM {0}
+                     JOIN {1} USING (response, survey_question_id)
+                     JOIN {2} USING (survey_question_id)
+                     LEFT JOIN {3} USING (survey_question_id)
+                     WHERE survey_id = %s AND retired = FALSE""".format(
+                self._survey_answers_table,
+                self._survey_question_response_table,
+                self._survey_question_response_type_table,
+                self._questions_table)
+            TRN.add(sql, [survey_id])
+            answers = TRN.execute_fetchindex()
+
+            TRN.add("""SELECT survey_question_id, response
+                       FROM {0}
+                       LEFT JOIN {1} using (survey_question_id)
+                       WHERE survey_id = %s AND retired = FALSE""".format(
+                self._survey_answers_other_table, self._questions_table),
+                [survey_id])
+            answers_other = TRN.execute_fetchindex()
+
+            survey = defaultdict(list)
+            for qid, idx, qtype in answers:
+                eid = self.questions[qid].interface_element_ids[0]
+                if qtype == 'SINGLE':
+                    survey[eid] = idx
+                else:
+                    survey[eid].append(idx)
+
+            for qid, data in answers_other:
+                eid = self.questions[qid].interface_element_ids[0]
+                data = str(data[2:-2])  # drop [""]
```

Comment:
There is a comment at the end of the line explaining what it does? I'm not sure what else I can do.